### ISSUE TITLE
Glusterfs server containers timing out

### ISSUE THREAD

**Issue by joostliketoast** [rancher/rancher#3750]
Rancher Version:

Rancher v0.59.1
Cattle  v0.148.1
User Interface  v0.90.0
Rancher Compose v0.7.2

Docker Version and OS:
![rancher](https://cloud.githubusercontent.com/assets/5398047/13498892/1aa86ca6-e15c-11e5-9d07-8d85ba518247.png)

Steps to Reproduce: 
start a glusterfs stack from servers on a internal network

Results: 
takes forever for everything to load for no apparent reason.
the containers keep rebooting in a loop because one of them always times out before the others finish starting. 

![rancher](https://cloud.githubusercontent.com/assets/5398047/13498967/88bb00aa-e15c-11e5-86c7-da983d88ab2c.png)
![rancher](https://cloud.githubusercontent.com/assets/5398047/13499025/c4fd1b0c-e15c-11e5-93cb-3db7c4ab80af.png)

Expected:

That server containers start quicker instead of so slowly. or that wait the timeout is set to longer than 30 seconds.

does hard disk size have any influence on starting time?

**Comment by joostliketoast** [comment]
![rancher](https://cloud.githubusercontent.com/assets/5398047/13499171/745d2ab0-e15d-11e5-8c70-eb3f66838a8c.png)

maybe this might also be of use

**Comment by joostliketoast** [comment]
okay, managed to get everything working took some effort for a glusterfs novice.
ill try and and describe the problem the best as i can with how how i solved the issue(a bit hacky though)

problem:

launching a glusterfs stack came into a reboot loop because the glusterfs-server containers took longer then 30 seconds to start. and wasn't able to get itself into a running mode

solution:

<img width="1408" alt="rancher" src="https://cloud.githubusercontent.com/assets/5398047/13521746/2718d946-e1eb-11e5-9ae2-890793b7bb31.png">

I had to manually time the start of the containers in box 1 to get them all running within the 30 second timeout.

After that finally worked and the stack was in running mode, I had the problem that convoy gluster couldn't boot. came across this ticket:
https://github.com/rancher/convoy/issues/56

which indeed there were no volumes running on the glusterfs stack.
so restart the containers highlighted in box 2.

and voila everything is running again.

TL;DR
gluster server containers start takes longer then 30 seconds, manually time restarts to get them running .
restart volume_create containers after that. then it works and convoy gluster should also not give mounting problems anymore.

**Comment by Negashev** [comment]
I think the problem rancher mysql. We have a rancher 20 stacks (~100 Conteners), after we have a server with SSD and rancher ran on it. all run at once. Rancher mysql become slower

**Comment by will-chan** [comment]
Closing due to inactivity, please reopen if you think this is still relevant to current releases & important.
